llama_index/llama_dataset/generator.py

```diff
@@ -0,0 +1,245 @@
+"""Dataset generation from documents."""
+from __future__ import annotations
+
+import asyncio
+import re
+from typing import List
+
+from llama_index import Document, ServiceContext, SummaryIndex
+from llama_index.ingestion import run_transformations
+from llama_index.llama_dataset import (
+    CreatedByType,
+    LabelledRagDataExample,
+    LabelledRagDataset,
+)
+from llama_index.llms.openai import OpenAI
+from llama_index.postprocessor.node import KeywordNodePostprocessor
+from llama_index.prompts.base import BasePromptTemplate, PromptTemplate
+from llama_index.prompts.default_prompts import DEFAULT_TEXT_QA_PROMPT
+from llama_index.prompts.mixin import PromptDictType, PromptMixin, PromptMixinType
+from llama_index.response.schema import RESPONSE_TYPE
+from llama_index.schema import BaseNode, MetadataMode, NodeWithScore
+
+DEFAULT_QUESTION_GENERATION_PROMPT = """\
+Context information is below.
+---------------------
+{context_str}
+---------------------
+Given the context information and not prior knowledge.
+generate only questions based on the below query.
+{query_str}
+"""
+
+
+def _get_default_service_context() -> ServiceContext:
+    """Get default service context."""
+    llm = OpenAI(temperature=0, model="gpt-3.5-turbo")
+    return ServiceContext.from_defaults(llm=llm, chunk_size_limit=3000)
+
+
+class RagDatasetGenerator(PromptMixin):
```

Comment:
Yeah, I spoke with Simon on this too. I think the preference would be to keep either `LabelledRagDataset` or `QueryResponseDataset`. In either case, have them live in `llama_dataset` module moving forward. My preference would be to retain `LabelledRagDataset`, which would mean we should mark both `QueryResponseDataset` and its generator as deprecated, as you suggested.

I'll do that now, and if no one feels strongly otherwise, then we can roll with this. And, in a future PR shortly after this one, I'll update docs that use the old stuff and update it to the new stuff.

Reply:
Nice, that sounds good!